Description: fix dkms-build on linux>=5.18
Author: Dimitri John Ledkov <dimitri.ledkov@canonical.com>
Origin: Ubuntu
Bug: https://github.com/umlaeute/v4l2loopback/issues/498
Last-Update: 2022-08-05
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- v4l2loopback.orig/dkms.conf
+++ v4l2loopback/dkms.conf
@@ -3,8 +3,15 @@
 
 if [ -f $kernel_source_dir/.config ]; then
     . $kernel_source_dir/.config
-    if [ "${CONFIG_VIDEO_V4L2:-n}" = "n" ]; then
-        BUILD_EXCLUSIVE_KERNEL="REQUIRES CONFIG_VIDEO_V4L2"
+    if ! { echo "$kernelver"; echo 5.18; } | sort -V -C; then
+        # for linux>=5.18, CONFIG_VIDEO_V4L2 has been renamed to CONFIG_VIDEO_DEV
+        if [ "${CONFIG_VIDEO_DEV:-n}" = "n" ]; then
+            BUILD_EXCLUSIVE_KERNEL="REQUIRES CONFIG_VIDEO_DEV"
+        fi
+    else
+        if [ "${CONFIG_VIDEO_V4L2:-n}" = "n" ]; then
+            BUILD_EXCLUSIVE_KERNEL="REQUIRES CONFIG_VIDEO_V4L2"
+        fi
     fi
 fi
 
